File: python/day-3/a.py
data = open('input.txt').read().split()
grid = [list(line) for line in data]
width = len(grid[0])

count = 0
pos = [0, 0]
for pos[0] in range(1, len(grid)):
    pos[1] = (pos[1] + 3) % width

    if grid[pos[0]][pos[1]] == '#':
        count += 1
        
print(count)

File: rust/day-3/a.rs
use std::fs;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
    let inp = fs::read_to_string("input.txt")?;

    let grid: Vec<Vec<char>> = inp
        .lines()
        .filter(|s| !s.is_empty())
        .map(|line| line.chars().collect())
        .collect();

    let width = grid[0].len();
    let mut count = 0;
    let mut pos = vec![0, 0];

    while pos[0] < grid.len() - 1 {
        pos[0] += 1;
        pos[1] = (pos[1] + 3) % width;

        if grid[pos[0]][pos[1]] == '#' {
            count += 1;
        }
    }

    println!("{}", count);
    Ok(())

}